velox/exec/HashProbe.cpp

```diff
@@ -280,6 +284,11 @@ void HashProbe::addInput(RowVectorPtr input) {
 
   deselectRowsWithNulls(hashers_, nonNullInputRows_);
 
+  if (table_->numDistinct() == 0) {
```

Comment:
I'm not sure I understand this fix. Would you explain?

Reply:
> when accessed later

Can you point me to that code? Does this happen only if there is an extra filter? If so, can be skip decoding inputs if there is no filter?

Reply:
Yes it's only happen with filter

Reply:
Got it. Would it make sense to do the extra processing (decoding of the input) only when the filter is present?

Reply:
Thanks and updated!